com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/TextureGenerator.java

```diff
@@ -187,22 +228,10 @@ else if (componentCount == 3)
         int texcCompressionType;
 
         int dataSize = width * height * 4;
-        ByteBuffer buffer = ByteBuffer.allocateDirect(dataSize);
-
-        // On Linux we run out of memory while trying to load a 4K texture.
-        // We split the pixel read out into blocks with 512 scan lines per block.
-        for (int y = 0; y < height; y+=512) {
 
-            int count = Math.min(height - y, 512);
 
-            int[] rasterData = new int[count*width*4];
-            image.getRaster().getPixels(0, y, width, count, rasterData);
-            for (int i = 0; i < rasterData.length; ++i) {
-                buffer.put((byte) (rasterData[i] & 0xff));
-            }
-        }
+        ByteBuffer buffer_input = getByteBuffer(image);
```

Comment:
The previous code was from 2015, and I'd like to think that the Linux machines are now all around beefier.

Also, even _before_ that fix, it was a big copy.
Now, it's just a "wrap" of the same memory, before we pass it into the texc.cpp.

We'll try this first, and if it becomes a problem, we'll look into it again.